parsing/parchmint-parser.js

```diff
@@ -204,6 +220,60 @@ class ParchmintParser {
                 compObj['entity'], ports);
     }
 
+    /**
+     * Parse a JSON object for the Connections.
+     *
+     * Fills the connections map with Connections where the keys are Layer IDs,
+     * and the values are arrays of Connections that exist on that Layer. Sets
+     * the parser invalid if a duplicate ID is found.
+     *
+     * @since 1.0.0
+     *
+     * @param {object} jsonObj  A parsed JSON object representing the Parchmint
+     *                          file.
+     */
+    parseConnections(jsonObj) {
+        jsonObj['connections'].forEach((connValue, index) => {
+            if (!this.isUniqueID(connValue['id'])) {
+                this.valid = false;
+                console.log('Parser: Duplicate ID (' + connValue['id'] + ') found in "connections" key. Skipping' +
+                        ' Connection with name ' + connValue['name'] + ' at index ' + index + '.');
```

Comment:
It would probably be more proper to call it the "connections array" rather than "connections key". Not sure if this change needs to be propagated elsewhere.

Reply:
One other thing we should check when printing the index (which I think we do in Inkwell as well and should probably also check) is that the order might get re-arranged after the JSON is read in. It might be better if we just print the entire slug of JSON that cause the problem and said something like "... was caused by the following section".

Reply:
The "key" has been changed to "array" in ```parseConnections``` and ```parseComponents```.

Reply:
The order seems to stay the same, that is, I'm getting the index I expect from that statement, but I like the idea of printing out the problematic JSON snippet. I'm going to have to write some kind of error/warning log system so we can output to the webpage. I'll try to implement the JSON printing there.

Reply:
Ok, open an issue about it and we can circle back to it then.

Reply:
For reference: #61.